### PR TITLE
Multithreading 01

### PR DESCRIPTION
**Pull request by Alex-Korzun**

### CODE REVIEW

**Review by vlykhvar — APPROVED** [review]

File: src/main/java/multithreading/practice/RunnableExample.java [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    private final Counter counter;
+
+    public RunnableExample(Counter counter) {
+        this.counter = counter;
+    }
+
+    @Override
+    public void run() {
+        System.out.println("Runnable example started working...");
+        for (int i = 0; i < 100; i++) {
```
Comment (RomanDubovskyi) [inline src/main/java/multithreading/practice/RunnableExample.java]: for the thread run you should increment counter field, not set I all the time new value

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/multithreading/practice/RunnableExample.java (hunk 1 above)
Comment (Alex-Korzun) [reply src/main/java/multithreading/practice/RunnableExample.java]: refactored

File: src/main/java/multithreading/practice/RunnableExample.java [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package multithreading.practice;
+
+import org.apache.log4j.Logger;
+
+public class RunnableExample implements Runnable {
+    private static final int ONE_HUNDRED = 100;
```
Comment (RomanDubovskyi) [inline src/main/java/multithreading/practice/RunnableExample.java]:
```suggestion
    private static final int COUNTING_LIMIT = 100;
```

File: src/main/java/multithreading/practice/RunnableExample.java [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    public RunnableExample(Counter counter) {
+        this.counter = counter;
+    }
+
+    @Override
+    public void run() {
+        logger.info("Runnable example started working...");
+        while (counter.getCounter() < ONE_HUNDRED) {
+            int count = counter.increment();
+            logger.info(Thread.currentThread().getName() + " = " + count);
```
Comment (RomanDubovskyi) [inline src/main/java/multithreading/practice/RunnableExample.java]:
```suggestion
            logger.info(Thread.currentThread().getName() + " = " + counter.increment());
```

File: src/main/java/multithreading/practice/ThreadExample.java [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+
+    public ThreadExample(Counter counter) {
+        this.counter = counter;
+    }
+
+    @Override
+    public void run() {
+        logger.info("Thread example started working...");
+        while (counter.getCounter() < ONE_HUNDRED) {
+            int count = counter.increment();
```
Comment (RomanDubovskyi) [inline src/main/java/multithreading/practice/ThreadExample.java]: same

**Review by RomanDubovskyi — CHANGES_REQUESTED** [review]

File: src/main/java/multithreading/practice/RunnableExample.java (hunk 2 above)
Comment (Alex-Korzun) [reply src/main/java/multithreading/practice/RunnableExample.java]: done

File: src/main/java/multithreading/practice/RunnableExample.java (hunk 3 above)
Comment (Alex-Korzun) [reply src/main/java/multithreading/practice/RunnableExample.java]: refactored

File: src/main/java/multithreading/practice/ThreadExample.java (hunk 4 above)
Comment (Alex-Korzun) [reply src/main/java/multithreading/practice/ThreadExample.java]: done

**Review by RomanDubovskyi — APPROVED** [review]

**Pull request merged**